### ISSUE TITLE
[FR] Custom URI commands

### ISSUE THREAD

**Issue by nathonius** [marcjulianschwarz/obsidian-raycast#55]
Using advanced URI makes tons of sense, but it feels like we're missing a lot of its power! It would be great to be able to configure custom URI commands.

For example, if I use Periodic Notes for my daily notes instead of the core daily notes plugin, it would be nice to be able to call the proper command from raycast.

I don't think that adding support for every single command/plugin is particularly sustainable, but allowing custom commands might be. 😄

**Comment by lemomar** [comment]
Bump!

**Comment by KevinBatdorf** [comment]
Can you comment a bit on how you would want this to work? Like a way to add a custom command + url combo to fire the url when the entry item is pressed?

**Comment by OfficerHalf** [comment]
> Can you comment a bit on how you would want this to work? Like a way to add a custom command + url combo to fire the url when the entry item is pressed? 

Exactly, yeah. Either a way to define an arbitrary URI as an option in raycast or or provide the ID of a command which the raycast plugin could format into the URI.

This all assumes I have a correct understanding of the advanced URI plugin. My understanding is that it allows you to call pretty much any command via URI.

**Comment by KevinBatdorf** [comment]
That sounds reasonable. Raycast won't be able to allow a dynamic user entered command, but I could create a generic command just for running custom commands. I'll have to investigate it some more though. I'll follow up.

**Comment by KevinBatdorf** [comment]
As far as I know, it grabs those from the package.json file, which runs through a validator before deploying to the store. So they need to be there beforehand. Could be some way though, I suppose.

**Comment by marcjulianschwarz** [comment]
If I remember correctly I was once talking with someone from Raycast about this and it was discouraged to have users create own commands. 
That is one reason why I for example used the Actions menu in the "Create Note" command to let users generate their own "Folder Actions".

So, imo like @KevinBatdorf suggested, a generic "URI command" would be the best solution.

**Comment by OfficerHalf** [comment]
Yeah, for sure. I'd be over the moon with that. However, I think you might be able to use either the Storage API or a file in the support directory with custom URIs, then show them in a [List](https://developers.raycast.com/api-reference/user-interface/list) or as [Actions](https://developers.raycast.com/api-reference/user-interface/actions#action).

But regardless a generic URI command is needed for that anyway, so this is a perfect place to start.
